processing/src/main/java/org/apache/carbondata/processing/loading/DataLoadProcessBuilder.java

```diff
@@ -222,8 +222,8 @@ public static CarbonDataLoadConfiguration createConfiguration(CarbonLoadModel lo
     configuration.setSegmentId(loadModel.getSegmentId());
     configuration.setTaskNo(loadModel.getTaskNo());
     configuration.setDataLoadProperty(DataLoadProcessorConstants.COMPLEX_DELIMITERS,
-        new String[] { loadModel.getComplexDelimiterLevel1(),
-            loadModel.getComplexDelimiterLevel2() });
+        new String[] { loadModel.getComplexDelimiterLevel1(), loadModel.getComplexDelimiterLevel2(),
+            loadModel.getComplexDelimiterLevel3(), loadModel.getComplexDelimiterLevel4() });
```

Comment:
Instead of adding these many delimeters methods in loadmodel, create a list or array of complex delimeters in loadmodel and then use it here

Reply:
done